grace/server.go

```diff
@@ -240,6 +248,7 @@ func (srv *Server) serverTimeout(d time.Duration) {
 			break
 		}
 		srv.wg.Done()
+		runtime.Gosched()
```

Comment:
StateTerminate is not protected by any mutex and not atomic, so the this for loop has a change not jump out, so the Done() will be called additionally, so the 'step' in waitGroup become -1